core/src/main/java/org/apache/calcite/sql/SqlSelect.java

```diff
@@ -66,8 +67,9 @@ public SqlSelect(SqlParserPos pos,
       SqlNode offset,
       SqlNode fetch,
       SqlNodeList hints) {
-    this(pos, keywordList, /*topN=*/ null, selectList, from, where, groupBy,
-        having, /*qualify=*/ null, windowDecls, orderBy, offset, fetch, hints);
+    this(pos, keywordList, /*topN=*/ null, /*exceptExpression=*/ null, selectList,
```

Comment:
From the constructor definition, it looks like exceptExpression actually comes after selectList.

Reply:
Very good catch, I think that may explain my build failure as well!